IPython/nbconvert/filters/markdown_mistune.py

```diff
@@ -23,7 +23,7 @@
 
 @undoc
 class MathBlockGrammar(mistune.BlockGrammar):
-    block_math = re.compile("^\$\$(.*?)\$\$", re.DOTALL)
+    block_math = re.compile("^\$\$([^\$]*?)\$\$", re.DOTALL)
```

Comment:
I am as stumped as you. I tried a few things, and this worked. I agree that it seems dark magic, but I don't know how many layers of regexen, parsers, lexers and whatnot are at war inside mistune, mathjax, jinja. There aren't a lot of places to check the internal state of mistune, in specific: very declarative.

Going back to good ol' print statements, it looks like blocks are hitting very infrequently, and a some markdown around `$` is getting ignored. I'll update the tests to fail accordingly!